tests/suites/test_suite_ssl.function

```diff
@@ -1556,6 +1589,241 @@ int exchange_data( mbedtls_ssl_context *ssl_1,
                                   ssl_2, 256, 1 );
 }
 
+#if defined(MBEDTLS_X509_CRT_PARSE_C)
+void perform_handshake( handshake_test_options* options )
```

Comment:
Should this return an `int`? How do the asserts work in this function? They don't have an "exit" label to go to, like their calling functions do.

Reply:
`perform_handshake` has an exit label. If a `TEST_ASSERT` fails, it not only goes to exit, but also calls `test_fail`, which is a helper function setting data about the fail in a static (file-scope) variable `test_info_t test_info;` which in turn is checked after each test case.